doc/memory.rst

```diff
@@ -345,6 +346,14 @@ Gotchas
         def method(self, ...):
 	    pass
 
+  Note that the first argument (`self`) will **not** be checked as
+  part of the caching. In other words, if you have two different
+  instances `foo1`, `foo2` which are both of type `Foo` and you call
+  `foo2.method` after `foo1.method` with the same arguments then the
+  second call will return the previously cached result, even though
+  `foo1` and `foo2` are different objects.
```

Comment:
If we want to go for a simple message to our non advanced users, what about a warning (using the warning directive in sphinx) saying that users are strongly encouraged to use @mem.cache on functions and not class methods?

IMHO if you insist on using @mem.cache on class methods other than static methods, I think you are just asking for trouble.